src/server/shell/basic_accessibility_manager.cpp

```diff
@@ -51,7 +51,8 @@ void mir::shell::BasicAccessibilityManager::repeat_delay(int new_delay) {
 }
 
 void mir::shell::BasicAccessibilityManager::notify_helpers() const {
-    for (auto const& helper: mutable_state.lock()->keyboard_helpers)
+    auto const keyboard_helpers = mutable_state.lock()->keyboard_helpers;
+    for (auto const& helper: keyboard_helpers)
         helper->repeat_info_changed(repeat_rate(), repeat_delay());
 }
```

Comment:
Isn't this the wrong change? What should the lock be protecting? Should we be locking and unlocking multiple times in a method call like this?

During the iteration we already have a lock on `mutable_state` giving safe access to `repeat_rate` and `repeat_delay`. The only problem is that it is an unnamed temporary, so extracting it seems better:

```c++
void mir::shell::BasicAccessibilityManager::notify_helpers() const {
    auto const state = mutable_state.lock();
    for (auto const& helper: state->keyboard_helpers)
        helper->repeat_info_changed(state.repeat_rate, repeat_delay);
}
```